context.xml configuration

bryanjbelanger opened the pull request:
This contains new manifests for context.xml configuration as well as a resource link.

mhaskel commented on manifests/config/context.pp:
```diff
@@ -0,0 +1,42 @@
+#   'absent'. Defaults to 'present'.
+# - The $port to wait for shutdown commands on.
+# - The $shutdown command that must be sent to $port.
+define tomcat::config::context (
```
See #66. I think this should probably have the option of either being in server.xml or in a separate file. The implementation in that PR has more flexibility also.

mhaskel commented on manifests/config/context.pp:
```diff
@@ -0,0 +1,42 @@
+#   address attribute. Valid values are 'true', 'false', 'present', or
+#   'absent'. Defaults to 'present'.
+# - The $port to wait for shutdown commands on.
+# - The $shutdown command that must be sent to $port.
```
Comments are copied from a different file

mhaskel commented on manifests/config/context.pp:
```diff
@@ -0,0 +1,42 @@
+  $address_ensure          = 'present',
+  $port                    = undef,
+  $shutdown                = undef,
+) {
```
These parameters aren't actually used.

mhaskel commented on manifests/config/context/resource.pp:
```diff
@@ -0,0 +1,68 @@
+#   beneath. Defaults to 'Catalina'.
+# - An optional hash of $additional_attributes to add to the Connector. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
```
These comments are all copied from a different file and not accurate

mhaskel commented on manifests/config/context/resource.pp:
```diff
@@ -0,0 +1,68 @@
+  $maxIdle,
+  $maxWait,
+  $url,
+  $factory,
```
It doesn't look like all these parameters are required. Anything that isn't actually required for the configuration of the Resource elements should be optional.

mhaskel commented on manifests/config/context/resourcelink.pp:
```diff
@@ -0,0 +1,48 @@
+#   beneath. Defaults to 'Catalina'.
+# - An optional hash of $additional_attributes to add to the Connector. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
```
These comments are not accurate.

mhaskel commented on manifests/config/context/resourcelink.pp:
```diff
@@ -0,0 +1,48 @@
+  $resource_link_name,
+  $type,
+  $catalina_base   = $::tomcat::catalina_home,
+  $resource_ensure = 'present',
```
This does not cover all the optional parameters for resource link elements

mhaskel commented on manifests/config/context/resourcelink.pp:
```diff
@@ -0,0 +1,48 @@
+# - An optional hash of $additional_attributes to add to the Connector. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
+define tomcat::config::context::resourcelink (
```
preference for this being named `tomcat::config::context::resource_link`

mhaskel commented on manifests/config/server/globalnamingresources.pp:
```diff
@@ -0,0 +1,126 @@
+#   beneath. Defaults to 'Catalina'.
+# - An optional hash of $additional_attributes to add to the Connector. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
```
These comments are inaccurate

mhaskel commented on manifests/config/server/globalnamingresources.pp:
```diff
@@ -0,0 +1,126 @@
+  $resource_name,
+  $auth,
+  $type,
+  $factory,
```
It doesn't look like all of these parameters are required for GlobalNamingResource elements

mhaskel commented on manifests/config/server/globalnamingresources.pp:
```diff
@@ -0,0 +1,126 @@
+  $driverClassName = undef,
+  $maxActive       = undef,
+  $maxIdle         = undef,
+  $maxWait         = undef,
```
These parameters should have underscores in the name not be camel cased for consistency

mhaskel commented:
@bryanjbelanger I've made some more specific comments inline, but for more general comments:

* I think ideally this PR would be broken up into several PRs as it's adding a bunch of new functionality that may be easier to merge in smaller pieces
* Each new class/define you add requires documentation updates (top of file and README)
* Each new class/define you add requires at least a new unit test
* Only parameters that are actually required per the tomcat documentation should be required in the manifests
* If implementing elements with several optional parameters considering doing something like https://github.com/puppetlabs/puppetlabs-tomcat/blob/master/manifests/config/server/connector.pp#L24-L25
* If any of these elements are supposed to be nested under parent elements make sure the parents are configurable
* PRs should be rebased/squashed down to at most one commit per feature (including tests/docs for that feature).

Let us know if you have any questions or need help addressing the comments.

bryanjbelanger commented:
Once I make changes how do I request a new pull?



On Jan 2, 2015, at 6:47 PM, Morgan Haskel <notifications@github.com<mailto:notifications@github.com>> wrote:


@bryanjbelanger<https://github.com/bryanjbelanger> I've made some more specific comments inline, but for more general comments:

  *   I think ideally this PR would be broken up into several PRs as it's adding a bunch of new functionality that may be easier to merge in smaller pieces
  *   Each new class/define you add requires documentation updates (top of file and README)
  *   Each new class/define you add requires at least a new unit test
  *   Only parameters that are actually required per the tomcat documentation should be required in the manifests
  *   If implementing elements with several optional parameters considering doing something like https://github.com/puppetlabs/puppetlabs-tomcat/blob/master/manifests/config/server/connector.pp#L24-L25
  *   If any of these elements are supposed to be nested under parent elements make sure the parents are configurable
  *   PRs should be rebased/squashed down to at most one commit per feature (including tests/docs for that feature).

Let us know if you have any questions or need help addressing the comments.

-
Reply to this email directly or view it on GitHub<https://github.com/puppetlabs/puppetlabs-tomcat/pull/71#issuecomment-68573924>.

mhaskel commented:
@bryanjbelanger you can either close this PR and just submit a new PR, or you can use rebasing with squashing and forced updates to your branch.

In general we recommend using the [Github Flow workflow](https://guides.github.com/introduction/flow/index.html)

bryanjbelanger commented:
Need to pull and resubmit.
